### PR TITLE
IO bomb: 

### PR DESCRIPTION
**Pull request by argonaut22**
- added quicksaves and quickloads of `aircraft`
- added `save_model` to save to output readable toml's a la `read_input`
- edited `read_input` to handle multi-mission and multi-point specs
- added `par_array_operations.jl` for convenient array manipulations (e.g., `array2nestedvector()`)
- added convenience aircraft import fxn's, `read_mdl()` and `read_jl()` in `read_externals.jl`
- added **`output_csv()`**. able to specify flight points, missions, par_array indices, to be written to csv. has logic for appending multiple aircraft to a single csv if outputs are consistent.
- added field to `aircraft` struct: `sized = [true/false]` as a sizing flag
- docstrings throughout (doc page and examples forthcoming)
- misc bugs and tweaks

### CODE REVIEW

**Comment by askprash** [comment]
@argonaut22 I just pulled in another PR #18 which have caused a few merge conflicts. Could you please resolve these?

**Comment by argonaut22** [comment]
Added tests. let's see what codecov says

File: example/example_drela_run.jl
Comment (askprash) [inline example/example_drela_run.jl:1]: Should we just get rid of this file?

File: src/TASOPT.jl [hunk 1]
```diff
@@ -33,10 +34,8 @@ export convertMass, convertForce, convertDist,
 
 include("./misc/index.inc")
 include("./misc/aircraft.jl")
-export aircraft
+export aircraft, fuselage_tank
```
Comment (askprash) [inline src/TASOPT.jl:37]: Nicer to put `fuselage_tank` in it's own file.

File: src/IO/read_externals.jl [hunk 2]
```diff
@@ -0,0 +1,43 @@
+include(joinpath(TASOPT.__TASOPTroot__, "misc/index.inc"))
```
Comment (askprash) [inline src/IO/read_externals.jl:1]: Initial thought was that we won't need these functions if we get rid of the `.jl` input files and the `.mdl` outputs but perhaps worth archiving somewhere?

File: Project.toml [hunk 3]
```diff
@@ -5,18 +5,23 @@ version = "1.0.0-DEV"
...
 Dierckx = "39dd38d3-220a-591b-8e3c-4c3a8c710a94"
 DocStringExtensions = "ffbed154-4ef7-542d-bbb7-c09d3a79fcae"
 ForwardDiff = "f6369f11-7733-5829-9624-2563aa707210"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 NLopt = "76087f3c-5699-56af-9a33-bf431cd00edd"
 NLsolve = "2774e3e8-f4cf-5e23-947b-6d7e65073b56"
+Parameters = "d96e819e-fc66-5662-9728-84c9c7592b0a"
 Printf = "de0858da-6303-5e67-8744-51eddeeeb8d7"
 PyPlot = "d330b81b-6aea-500a-939a-2ce795aea3ee"
+Revise = "295af30f-e4ad-537b-8983-00126c2a3abe"
```
[the remark this answers is not in the document]
Comment (argonaut22) [reply Project.toml:20]: agreed. done

File: Project.toml [hunk 4]
```diff
@@ -5,18 +5,23 @@ version = "1.0.0-DEV"
...
+CSV = "336ed68f-0bac-5ca0-87d4-7b16caf5d00b"
 Dates = "ade2ca70-3891-5945-98fb-dc099432e06a"
+DelimitedFiles = "8bb1440f-4735-579b-a4ab-409b98df4dab"
 Dierckx = "39dd38d3-220a-591b-8e3c-4c3a8c710a94"
 DocStringExtensions = "ffbed154-4ef7-542d-bbb7-c09d3a79fcae"
 ForwardDiff = "f6369f11-7733-5829-9624-2563aa707210"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 NLopt = "76087f3c-5699-56af-9a33-bf431cd00edd"
 NLsolve = "2774e3e8-f4cf-5e23-947b-6d7e65073b56"
+Parameters = "d96e819e-fc66-5662-9728-84c9c7592b0a"
```
[the remark this answers is not in the document]
Comment (argonaut22) [reply Project.toml:17]: i actually used this for @unpack, which is actually in UnPack.jl. either way. deleted

File: src/IO/IO_samples/777TFinput_edited.jl
[the remark this answers is not in the document]
Comment (argonaut22) [reply src/IO/IO_samples/777TFinput_edited.jl:1]: these are the old .mdl models with the tweaks needed to make it run. forgot they were in there

This one in particular is used for the `read_mdl()` default/example. could just take out the default. your call

File: example/example_drela_run.jl
Comment (argonaut22) [reply example/example_drela_run.jl:1]: looks like past me agrees with you but lacked confidence. deleting now

image

[the remark this answers is not in the document]
Comment (argonaut22) [reply example/example_drela_run.jl:1]: agreed and deleted

File: example/example_TF_run.jl
[the remark this answers is not in the document]
Comment (argonaut22) [reply example/example_TF_run.jl:1]: agreed. deleted

File: src/misc/aircraft.jl [hunk 5]
```diff
@@ -1,3 +1,8 @@
+"""
+$TYPEDEF
+Fuselage tank component. Usually for Hydrogen aircraft
+$TYPEDFIELDS
+"""
 mutable struct fuselage_tank
```
[the remark this answers is not in the document]
Comment (argonaut22) [reply src/misc/aircraft.jl:6]: i agree, but vote to fix it with the other struct changes we need to make

File: src/TASOPT.jl (hunk 1 above)
Comment (argonaut22) [reply src/TASOPT.jl:37]: ^

**Review by askprash — APPROVED** [review]

**Pull request merged**